Cod3rsGrowth.forms/FormListagem.cs

```diff
@@ -267,27 +277,16 @@ private void AoClicarNoBotaoEditarVenda(object sender, EventArgs e)
             try
             {
                 var tabelaVenda = _servicoVenda.ObterTodos(_filtroVenda);
```

Comment:
Aqui você faz um ObterTodos para saber se tem algum item no repositorio.
Para depois interagir com sua TabelaVenda e pegar o id.
VocÊ pode remover esse ObterTodos e veirificar diretamente na TabelaVenda se TabelaVenda.Rows é diferente de null, porque se for é porque tem algum valor adicionado ao seu dataGrid.